src/kepler_model/util/similarity.py

```diff
@@ -0,0 +1,89 @@
+from .train_types import NodeAttribute
+
+# from similarity result on SPEC data
+similarity_reference = {
+    NodeAttribute.PROCESSOR: 0.56,
+    NodeAttribute.CORES: 0.76,
+    NodeAttribute.CHIPS: 0.52,
+    NodeAttribute.MEMORY: 0.8,
+    NodeAttribute.FREQ: 0.01,
+}
+
+thread_related_weight = 0.76 + 0.52 + 0.12
+similarity_reference = {
+    NodeAttribute.PROCESSOR: 3,
+    NodeAttribute.CORES: 0.76/thread_related_weight,
+    NodeAttribute.CHIPS: 0.52/thread_related_weight,
+    NodeAttribute.MEMORY: 0.8,
+    NodeAttribute.FREQ: 0.01,
+}
+
+similarity_total_weight = sum(similarity_reference.values())
+
+def get_similarity_weight(attr):
+    return similarity_reference[attr]/similarity_total_weight
+
+def compute_jaccard_similarity(str1, str2):
+    if str1.lower() == str2.lower():
+        return 1
+    set1 = set(str1.lower())  # Convert to lowercase for case-insensitive comparison
+    set2 = set(str2.lower())
+
+    intersection = len(set1.intersection(set2))
+    union = len(set1.union(set2))
+
+    similarity = intersection / union if union != 0 else 0
+    return similarity/2
```

Comment:
```suggestion
    similarity = intersection / union * 0.5
    return similarity
```
Given that `union` can't be 0 because of len() != 0 check above.